cpp/tests/table/experimental_row_operator_tests.cu

```diff
@@ -0,0 +1,300 @@
+                     Comparator c)
+{
+  rmm::cuda_stream_view stream{};
+
+  auto table_comparator = lexicographic::self_comparator{input, column_order, {}, stream};
+  auto comparator       = table_comparator.less(cudf::nullate::NO{}, c);
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator(0),
+                    thrust::make_counting_iterator(input.num_rows()),
+                    thrust::make_counting_iterator(0),
+                    output.data<int8_t>(),
+                    comparator);
+}
+
+template <typename Comparator>
+void row_equality(cudf::table_view input1,
+                  cudf::table_view input2,
+                  cudf::mutable_column_view output,
+                  std::vector<cudf::order> const& column_order,
+                  Comparator c)
+{
+  rmm::cuda_stream_view stream{};
+
+  auto table_comparator = equality::two_table_comparator{input1, input2, stream};
+  auto comparator   = table_comparator.equal_to(cudf::nullate::NO{}, cudf::null_equality::EQUAL, c);
+  auto const lhs_it = cudf::experimental::row::lhs_iterator(0);
+  auto const rhs_it = cudf::experimental::row::rhs_iterator(0);
+  thrust::transform(rmm::exec_policy(stream),
+                    lhs_it,
+                    lhs_it + input1.num_rows(),
+                    rhs_it,
+                    output.data<bool>(),
+                    comparator);
+}
+
+template <typename Comparator>
+void self_equality(cudf::table_view input,
+                   cudf::mutable_column_view output,
+                   std::vector<cudf::order> const& column_order,
+                   Comparator c)
+{
+  rmm::cuda_stream_view stream{};
+
+  auto table_comparator = equality::self_comparator{input, stream};
+  auto comparator = table_comparator.equal_to(cudf::nullate::NO{}, cudf::null_equality::EQUAL, c);
+  thrust::transform(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator(0),
+                    thrust::make_counting_iterator(input.num_rows()),
+                    thrust::make_counting_iterator(0),
+                    output.data<bool>(),
+                    comparator);
+}
+
+TYPED_TEST(TypedTableViewTest, EmptyColumnedTable)
+{
+  std::vector<cudf::column_view> cols{};
+
+  cudf::table_view input(cols);
+  cudf::size_type expected = 0;
+
+  EXPECT_EQ(input.num_columns(), expected);
+}
+
+TYPED_TEST(TypedTableViewTest, TestLexicographicalComparatorTwoTableCase)
+{
+  using T = TypeParam;
+
+  fixed_width_column_wrapper<T> col1{{1, 2, 3, 4}};
+  fixed_width_column_wrapper<T> col2{{0, 1, 4, 3}};
+  std::vector<cudf::order> column_order{cudf::order::DESCENDING};
+
+  cudf::table_view input_table_1{{col1}};
+  cudf::table_view input_table_2{{col2}};
+
+  auto got = cudf::make_numeric_column(
+    cudf::data_type(cudf::type_id::INT8), input_table_1.num_rows(), cudf::mask_state::UNALLOCATED);
+  fixed_width_column_wrapper<int8_t> expected{{1, 1, 0, 1}};
+
+  row_comparison(input_table_1,
```

Comment:
These tests would be a lot cleaner if the `row_comparison` returned a `std::unique_ptr<column>` (like most cudf algorithms) of `bool` dtype instead of accepting/mutating a mutable view. Move the `make_numeric_column` logic into the algorithm implementation rather than creating the object in each test.

Result would look like:
```cpp
auto const got = row_comparison(input_table_1, input_table_2, column_order, lexicographic::physical_element_comparator{});
```

Reply:
As a note, most of the functions I can think of that accept/mutate mutable views are doing so because they're a direct wrapper around a Thrust or CUB call that takes raw iterators, and they usually live in a detail namespace. User-facing algorithms generally allocate and return owning objects like `std::unique_ptr<column>`.

Reply:
I overhauled this file in 3a7f118. I need to make sure it compiles and will push again soon.